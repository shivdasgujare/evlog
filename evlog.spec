#
# Enterprise Event Log Spec
#
Summary:   Enterprise Event Logging
Name:      evlog
Version:   1.6.1 
Release:   1
Copyright: LGPL
Vendor:    IBM <http://www.ibm.com>
Url:       http://evlog.sourceforge.net

Packager:  Hien Nguyen <nguyhien@us.ibm.com>
Group:     System Environment/Base
Source:    evlog-1.6.1.tar.gz
BuildRoot: /var/tmp/%{name}-buildroot

%description
Enterprise Event Logging

%prep
%setup
CFLAGS="$RPM_OPT_FLAGS" CXXFLAGS="$RPM_OPT_FLAGS" $LOCALFLAGS
./configure 

%build
make -C user

%install
rm -rf $RPM_BUILD_ROOT
make DESTDIR=$RPM_BUILD_ROOT install

# Copy ela* stuff to /usr/bin for now, we will use move later.
mkdir -p $RPM_BUILD_ROOT/usr/sbin
cp $RPM_BUILD_ROOT/sbin/ela_* $RPM_BUILD_ROOT/usr/sbin

cd $RPM_BUILD_ROOT
find . -type d | sed '1,2d;s,^\.,\%attr(-\,root\,root) \%dir ,' |grep -v '/etc/init.d' > $RPM_BUILD_DIR/file.list.evlog-user
find . -type f | sed 's,^\.,\%attr(-\,root\,root) ,' >> $RPM_BUILD_DIR/file.list.evlog-user
find . -type l | sed 's,^\.,\%attr(-\,root\,root) ,' >> $RPM_BUILD_DIR/file.list.evlog-user

%pre
#echo This is pre arg=$1
if [ "$1" = "1" -o "$1" = "2" ] ; then
	if [ -f /etc/init.d/evlaction ]
	then
		PID3=`ps -ef | grep evlactiond | grep -v grep | awk '{print $2}'`
		test -z $PID3 || /etc/init.d/evlaction stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlnotify ]
	then
		PID2=`ps -ef | grep evlnotifyd | grep -v grep | awk '{print $2}'`
		test -z $PID2 || /etc/init.d/evlnotify stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlogrmtd ]
	then
		PID1=`ps -ef | grep evlogrmtd | grep -v grep | awk '{print $2}'`
		test -z $PID1 || /etc/init.d/evlogrmtd stop
		/bin/sleep 1
	fi	
	if [ -f /etc/init.d/evlog ]
	then
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		test -z $PID || /etc/init.d/evlog stop
		/bin/sleep 1
	fi	

	# We don't use logrotate anymore so remove evlog under logrotate.d 
	if [ -f /etc/logrotate.d/evlog ]
	then
		rm -f /etc/logrotate.d/evlog
	fi	
	
	if [ -d /etc/rc.d/init.d ]
	then 
		# Redhat style, make a link to /etc/init.d
		test -L /etc/init.d || ln -s /etc/rc.d/init.d /etc/init.d
	fi
	
	if [ -d /etc/init.d/rc3.d ]
	then
		# Suse style of rc0-6.d
		test -L /etc/rc.d || ln -s /etc/init.d /etc/rc.d
	fi
fi
	
%post
#echo This is post arg=$1
if [ "$1" = "1" -o "$1" = "2" ] ; then
	/sbin/ldconfig

	(cd /var/evlog/templates; make)
	
	/etc/init.d/evlnotify start
	/bin/sleep 1
	/etc/init.d/evlaction start
	/bin/sleep 1
	/etc/init.d/evlog start
	/bin/sleep 1
	/etc/init.d/evlogrmt start
	
	REL_INITDOTD_DIR=../../init.d
	RCDOTD_BASE=/etc/rc.d

	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc3.d/S09evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc5.d/S09evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc0.d/K90evlog
	ln -fs $REL_INITDOTD_DIR/evlog $RCDOTD_BASE/rc6.d/K90evlog

	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc3.d/S89evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc5.d/S89evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc0.d/K60evlogrmt
	ln -fs $REL_INITDOTD_DIR/evlogrmt $RCDOTD_BASE/rc6.d/K60evlogrmt
	
	# Remove previous version (1.5.3 and earlier) links if exist
	test ! -L $RCDOTD_BASE/rc3.d/S80evlnotify || rm -f $RCDOTD_BASE/rc3.d/S80evlnotify
	test ! -L $RCDOTD_BASE/rc5.d/S80evlnotify || rm -f $RCDOTD_BASE/rc5.d/S80evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc3.d/S07evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc5.d/S07evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc0.d/K50evlnotify
	ln -fs $REL_INITDOTD_DIR/evlnotify $RCDOTD_BASE/rc6.d/K50evlnotify
	
	# Remove previous version (1.5.3 and earlier) links if exist
	test ! -L $RCDOTD_BASE/rc3.d/S90evlaction || rm -f $RCDOTD_BASE/rc3.d/S90evlaction
	test ! -L $RCDOTD_BASE/rc5.d/S90evlaction || rm -f $RCDOTD_BASE/rc5.d/S90evlaction

	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc3.d/S08evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc5.d/S08evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc0.d/K40evlaction
	ln -fs $REL_INITDOTD_DIR/evlaction $RCDOTD_BASE/rc6.d/K40evlaction
fi


%preun
#echo This is preun arg=$1
if [ "$1" = "0" ] ; then

	if [ -f /etc/init.d/evlaction ]
	then
		PID3=`ps -ef | grep evlactiond | grep -v grep | awk '{print $2}'`
		test -z $PID3 || /etc/init.d/evlaction stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlnotify ]
	then
		PID2=`ps -ef | grep evlnotifyd | grep -v grep | awk '{print $2}'`
		test -z $PID2 || /etc/init.d/evlnotify stop
		/bin/sleep 1
	fi
	if [ -f /etc/init.d/evlogrmt ]
	then
		PID1=`ps -ef | grep evlogrmtd | grep -v grep | awk '{print $2}'`
		test -z $PID1 || /etc/init.d/evlogrmt stop
		/bin/sleep 1
	fi	
	if [ -f /etc/init.d/evlog ]
	then
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		test -z $PID || /etc/init.d/evlog stop
		/bin/sleep 1
	fi	

	(cd /var/evlog/templates; make clobber)

	#
	# remove libevlsyslog.so.1 entry in the /etc/ld.so.preload if any
	#
	/sbin/slog_fwd -r
fi
	
%postun
#echo This is postun arg=$1
if [ "$1" = "0" ] ; then
	#
	# remove startup scripts link
	#

	if [ -e /etc/init.d/evlog ]
	then
		rm -f /etc/init.d/evlog
	fi
	if [ -e /etc/init.d/evlogrmt ]
	then
		rm -f /etc/init.d/evlogrmt
	fi
	if [ -e /etc/init.d/evlnotify ]
	then
		rm -f  /etc/init.d/evlnotify
	fi
	if [ -e /etc/init.d/evlaction ]
	then
		rm -f /etc/init.d/evlaction
	fi 
	
	RCDOTD_BASE=/etc/rc.d

	rm -f $RCDOTD_BASE/rc3.d/S09evlog
	rm -f $RCDOTD_BASE/rc5.d/S09evlog
	rm -f $RCDOTD_BASE/rc0.d/K90evlog
	rm -f $RCDOTD_BASE/rc6.d/K90evlog

	rm -f $RCDOTD_BASE/rc3.d/S89evlogrmt
	rm -f $RCDOTD_BASE/rc5.d/S89evlogrmt
	rm -f $RCDOTD_BASE/rc0.d/K60evlogrmt
	rm -f $RCDOTD_BASE/rc6.d/K60evlogrmt

	rm -f $RCDOTD_BASE/rc3.d/S07evlnotify
	rm -f $RCDOTD_BASE/rc5.d/S07evlnotify
	rm -f $RCDOTD_BASE/rc0.d/K50evlnotify
	rm -f $RCDOTD_BASE/rc6.d/K50evlnotify

	rm -f $RCDOTD_BASE/rc3.d/S08evlaction
	rm -f $RCDOTD_BASE/rc5.d/S08evlaction
	rm -f $RCDOTD_BASE/rc0.d/K40evlaction
	rm -f $RCDOTD_BASE/rc6.d/K40evlaction

	/sbin/ldconfig	
fi
	

%clean
rm -rf $RPM_BUILD_ROOT/*
rm -rf $RPM_BUILD_DIR/evlog
rm -rf ../file.list.evlog


%files -f ../file.list.evlog-user

%changelog
* Tue Apr 27 2004 Hien Nguyen <nguyhien@us.ibm.com>
- Release version of v1.6.0. Added ELA feature.
* Thu Dec 18 2003 Hien Nguyen <nguyhien@us.ibm.com>
- Added ELA feature - and bugs fixed
* Mon Jun 16 2003 Hien Nguyen <nguyhien@us.ibm.com>
- Add multi-arch support
* Thu Mar 7 2002 Jim Keniston <kenistoj@us.ibm.com>
- Added kernel module tests kfacreg and kfacreg2 (test3, test4)
* Fri Oct 12 2001 Hien Nguyen <nguyhien@us.ibm.com>
- Bug fixes, add template, evltc, evlfacility, libevlsyslog
  and many small enhancements.
* Fri Sep 28 2001 Hien Nguyen <nguyhien@us.ibm.com>
- Bug fixes, adding s390 and s390x specific tests 
* Tue Aug 26 2001 Hien Nguyen <nguyhien@us.ibm.com>
- First RPM build
