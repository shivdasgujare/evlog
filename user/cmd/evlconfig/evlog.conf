#
# Configuration values for evlogd daemon. 
# Use 'evlconfig' command to update these values. If this file is edited 
# manually, the updated value will not be used by the evlogd daemon. So,
# the evlogd daemon has to be restarted. Hence, use 'evlconfig' command
# to send this values to evlogd daemon.
# Console display level - OFF=111
#
Discard Dups: on
Discard Interval: 3
Discard Count: 100
Event Screen: nofilter
Console display level: 111
Lookback Size: 3
