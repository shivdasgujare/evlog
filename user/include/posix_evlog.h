/*
 * Linux Event Logging for the Enterprise
 * Copyright (c) International Business Machines Corp., 2001
 *
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 *  Please send e-mail to lkessler@users.sourceforge.net if you have
 *  questions or comments.
 *
 *  Project Website:  http://evlog.sourceforge.net/
 *
 */

#define LOG_LOGMGMT (12<<3) /* this facility defined by POSIX, but not syslog */
#ifndef _POSIX_EVLOG_H_
#define _POSIX_EVLOG_H_

#include <sys/types.h>
/* For sigevent */
#include <signal.h>
/* For facility (e.g., LOG_KERN) and severity (e.g., LOG_EMERG) codes */
#include <syslog.h>
/* For va_list */
#include <stdarg.h>

#ifdef __cplusplus
extern "C" {
#endif	/* __cplusplus */

#ifdef _POSIX_THREADS
#include <pthread.h>
#endif
#include <linux/evl_log.h>

/* Treat a log descriptor like a file descriptor for now. */
typedef struct _log_description *posix_logd_t;

/* Member selectors for posix_log_entry, for use by posix_log_memtostr  */
#define POSIX_LOG_ENTRY_DATA	0	/* variable-length data */
#define POSIX_LOG_ENTRY_RECID	1
#define POSIX_LOG_ENTRY_SIZE	2
#define POSIX_LOG_ENTRY_FORMAT	3
#define POSIX_LOG_ENTRY_EVENT_TYPE	4
#define POSIX_LOG_ENTRY_FACILITY	5
#define POSIX_LOG_ENTRY_SEVERITY	6
#define POSIX_LOG_ENTRY_UID		7
#define POSIX_LOG_ENTRY_GID		8
#define POSIX_LOG_ENTRY_PID		9
#define POSIX_LOG_ENTRY_PGRP	10
#define POSIX_LOG_ENTRY_TIME	11
#define POSIX_LOG_ENTRY_FLAGS	12
#define POSIX_LOG_ENTRY_THREAD	13
#define POSIX_LOG_ENTRY_PROCESSOR	14

/* Query purpose flags */
#define POSIX_LOG_PRPS_NOTIFY	0x1
#define POSIX_LOG_PRPS_GENERAL	0x2
#define POSIX_LOG_PRPS_SEEK	(POSIX_LOG_PRPS_GENERAL)
#define EVL_PRPS_TEMPLATE	0x4
#define EVL_PRPS_RESTRICTED	0x8
#define POSIX_LOG_PRPS_MASK	0xf

struct node;

typedef struct _posix_log_query {
	int		qu_purpose;	/* NOTIFY, SEEK, etc. */
	char		*qu_expr;	/* Saved copy of the query expr */
	struct node	*qu_tree;	/* Parse tree */
	struct evl_nonStdAtts	*qu_nonStdAtts;	/* Non-standard attributes. */
} posix_log_query_t;


#define POSIX_LOG_ONCE_ONLY		0x01
#define POSIX_LOG_SEND_RECID		0x02
#define POSIX_LOG_SEND_SIGVAL		0x04
#define POSIX_LOG_NFY_MASK		0x07

#define POSIX_LOG_NFY_DISABLED		0x08
#define POSIX_LOG_ACTION_PERSIST	0x16

#define POSIX_LOG_NFY_PENDING_REMOVE	0x512

#define SI_EVLOG	SI_QUEUE
typedef size_t posix_log_notify_t;
/*
 * These are the types of messages that can be sent between the client (a
 * process that has called posix_log_notify_add_ex()) and the server (the
 * notification daemon).
 */
typedef enum _nfyMsgType {
	nmtNewRequest,			/* new notification request */
	nmtGetRequestStatus,	/* get status of notification request */
	nmtRmRequest,			/* remove notification request */
	nmtRequestStatus		/* response to client requests */
} notifyMsgType_t;

/*
 * The status of a notification request.
 */
typedef enum _nfyRqStatus {
	nrsEnabled,		/* Request has been accepted by server. */
	nrsDisabled,	/* Once-only request has been satisfied */
					/* by the server, but not removed by the */
					/* client. */
	nrsRemoved,		/* Request has been removed at client's request. */
	nrsNoRequest	/* Request not found by server */
} nfyRqStatus_t;

/*
 * The header of a message between client and server.
 */
typedef struct _nfyMsgHdr {
	int	nhMsgType;
	posix_log_notify_t	nhReqHandle;
	int	nhStatus;
} nfyMsgHdr_t;

typedef struct _nfyNewRqMsg {
	nfyMsgHdr_t	nwHeader;
	size_t		nwQueryLength;
	struct sigevent	nwSigevent;
	int		nwFlags;
} nfyNewRqMsg_t;

#define EVL_NFY_MSGMAXSIZE (sizeof(nfyNewRqMsg_t))

/* Log-management event-type codes */
#define POSIX_LOG_MGMT_TIMEMARK	1
#define POSIX_LOG_MGMT_STARTMAINT	2
#define POSIX_LOG_MGMT_ENDMAINT	3
#define POSIX_LOG_MGMT_STOPLOGGING	4
#define POSIX_LOG_MGMT_STARTLOGGING	5

/* Per-process maximum number of concurrent open log descriptors */
#define POSIX_LOG_OPEN_MAX	100

/* Per-process max number of concurrently registered notification requests */
#define POSIX_LOG_NOTIFY_MAX	100

/* Seek directions */
#define POSIX_LOG_SEEK_START	1
#define POSIX_LOG_SEEK_END		2
#define POSIX_LOG_SEEK_FORWARD	3
#define POSIX_LOG_SEEK_BACKWARD	4
#define POSIX_LOG_SEEK_FIRST	5
#define POSIX_LOG_SEEK_LAST		6

/* Maximum length for a string returned by posix_log_memtostr */
#define POSIX_LOG_MEMSTR_MAXLEN	128

/* Log an event */
extern int posix_log_write(posix_log_facility_t facility,
	int event_type, posix_log_severity_t severity,
	const void *buf, size_t len, int format, unsigned int flags);

extern int posix_log_vprintf(posix_log_facility_t facility,
	int event_type, posix_log_severity_t severity, unsigned int flags,
	const char *format, va_list args);

extern int posix_log_printf(posix_log_facility_t facility,
	int event_type, posix_log_severity_t severity, unsigned int flags,
	const char *format, ...);

extern int posix_log_vprintb(posix_log_facility_t facility,
	int event_type, posix_log_severity_t severity, unsigned int flags,
	const char *format, va_list args);

extern int posix_log_printb(posix_log_facility_t facility,
	int event_type, posix_log_severity_t severity, unsigned int flags,
	const char *format, ...);

/* Open an event log for read access */
extern int posix_log_open(posix_logd_t *logdes, const char *path);

/* Read from event log */
extern int posix_log_read(posix_logd_t logdes, struct posix_log_entry *entry,
	void *log_buf, size_t log_len); 

/* Register for notification about new events */
extern int posix_log_notify_add(const posix_log_query_t *query,
	const struct sigevent *notification, int flags,
	posix_log_notify_t *nfyhandle);

extern int posix_log_sigval_getrecid(const union sigval sval,
	posix_log_recid_t *recid);

extern int posix_log_siginfo_getrecid(const siginfo_t *info,
        void *context, posix_log_recid_t *recid);
        
extern int posix_log_notify_get(posix_log_notify_t nfyhandle,
	struct sigevent *notification, int *flags,
	char *qsbuf, size_t qslen, size_t *reqlen);

/* Remove notification request */
extern int posix_log_notify_remove(posix_log_notify_t nfyhandle);

/* Close event log */
extern int posix_log_close(posix_logd_t logdes);

/* Reposition the read pointer */
extern int posix_log_seek(posix_logd_t logdes, const posix_log_query_t *query,
	int direction); 

/* Compare severities */
extern int posix_log_severity_compare(int *order, posix_log_severity_t s1,
	posix_log_severity_t s2); 

/* Create log query */
extern int posix_log_query_create(const char *query_string, int purpose,
	posix_log_query_t *query, char *errbuf, size_t errlen);

extern int posix_log_query_get(const posix_log_query_t *query, int *purpose,
	char *qsbuf, size_t qslen, size_t *reqlen);

/* Destroy log query */
extern int posix_log_query_destroy(posix_log_query_t *query);

/* Obtain the string equivalent of an event-record member */
extern int posix_log_memtostr(int member_selector,
	const struct posix_log_entry *entry,
	char *buf, size_t buflen);

/* Convert string to facility or facility to string */
extern int posix_log_factostr(posix_log_facility_t fac, char *buf,
	size_t buflen);

extern int posix_log_strtofac(const char *str, posix_log_facility_t *fac);

#ifdef __cplusplus
}
#endif 	/* __cplusplus */
#endif	/* _POSIX_EVLOG_H_ */
