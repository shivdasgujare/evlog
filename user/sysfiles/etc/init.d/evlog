#!/bin/sh
#
# chkconfig: 35 09 90
# description: Enterprise event log daemon
#
#

# Source function library.

REDHAT=`cat /proc/version | grep "Red Hat"`
# Source function library.
if [ ! -z "$REDHAT" ]
then
	. /etc/init.d/functions
fi


[ -f /sbin/evlogd ] || exit 0

EVLOGD_OPTIONS=""
# Source config
if [ -f /etc/sysconfig/evlog ]; then
	. /etc/sysconfig/evlog
fi

RETVAL=0

umask 077

record_last_recid() {
	/sbin/evlview -l $1 -t 1 -S %recid% > $2 2> /dev/null
	test -s $2 || echo 0 > $2
	chmod 644 $2
}

start() {
 	echo -n $"Starting enterprise event logger: "

	record_last_recid /var/evlog/eventlog /var/evlog/bootrecid
	record_last_recid /var/evlog/privatelog /var/evlog/pbootrecid
 	
 	if [ ! -z "$REDHAT" ]
	then 	
		daemon /sbin/evlogd $EVLOGD_OPTIONS
	else
		/sbin/evlogd $EVLOGD_OPTIONS
	fi
	sleep 1
	RETVAL=$?
	echo
	touch /var/lock/subsys/evlog
	return $RETVAL
}	
stop() {
	echo -n $"Shutting down enterprise event logger: "
	
	if [ ! -z "$REDHAT" ]
	then
		killproc evlogd
	else
		PID=`ps -ef | grep evlogd | grep -v grep | awk '$3==1 {print $2}'`
		if [ ! -z "$PID" ]
		then
			kill -s SIGTERM $PID
		fi
	fi

	RETVAL=$?
	echo
	rm -f /var/lock/subsys/evlog
	if [ -e /var/run/evlogd.pid ]
	then
		rm -f /var/run/evlogd.pid
	fi
	return $RETVAL
}
rhstatus() {
	status evlogd
}
restart() {
	stop
	start
}	

case "$1" in
  start)
  	start
	;;
  stop)
  	stop
	;;
  status)
  	rhstatus
	;;
  restart|reload)
  	restart
	;;
  condrestart)
    restart	
	;;
  *)
	echo $"Usage: $0 {start|stop|status|restart|condrestart}"
	exit 1
esac

exit $?

