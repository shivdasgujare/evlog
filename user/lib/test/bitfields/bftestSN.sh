#!/bin/bash
#
TEST_TEMPLATE=bftestSN.out

EVLTMPLPATH=$EVL_TESTDIR/templates
export EVLTMPLPATH

if [ "$USER" = root ]
then
	/sbin/evltc -c $EVLTMPLPATH/user/bftestS.t
else
echo >&2 "bftestSN.sh: Not running as root.  Assuming template source file"
echo >&2 "$EVLTMPLPATH/user/bftestS.t has already been compiled."
fi

logtime=`date +%s`
sleep 1
./bftestSN
sleep 1
/sbin/evlview -f "(time>${logtime} && event_type>53000 && event_type<53200)" -S '%event_type:d%: %data%' > bftestSN.sh.out

diff bftestSN.sh.out $TEST_TEMPLATE

if [ $? -eq "0" ]
then
	echo templates bitfield test signed,negative	:PASSED
else
	echo templates bitfield test signed,negative	:FAILED
fi
