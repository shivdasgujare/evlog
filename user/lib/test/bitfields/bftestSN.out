53001: natts = 3 	-1 -1 -1 
53002: natts = 3 	-1 -1 -1 
53003: natts = 2 	-1 -1 
53004: natts = 5 	-1 -1 -1 -1 -1 
53005: natts = 5 	-1 -1 -1 -1 -1 
53006: natts = 4 	-1 -1 -1 -1 
53007: natts = 4 	-1 -1 -1 -1 
53008: natts = 3 	-1 -1 -1 
53010: natts = 5 	-1 -1 -1 -1 -1 
53011: natts = 5 	-1 -1 -1 -1 -1 
53012: natts = 5 	-1 -1 -1 -1 -1 
53013: natts = 5 	-1 -1 -1 -1 -1 
53014: natts = 3 	-1 -1 -1 
53015: natts = 3 	-1 -1 -1 
53016: natts = 3 	-1 -1 -1 
53017: natts = 3 	-1 -1 -1 
53018: natts = 3 	-1 -1 -1 
53019: natts = 3 	-1 -1 -1 
53020: natts = 3 	-1 -1 -1 
53021: natts = 6 	-1 -1 -1 -1 -1 -1 
53121: natts = 6 	-1 -1 -1 -1 -1 -1 
53022: natts = 3 	-1 -1 -1 
53023: natts = 3 	-1 -1 -1 
53024: natts = 7 	-1 -1 -1 -1 -1 -1 -1 
53124: natts = 7 	-1 -1 -1 -1 -1 -1 -1 
53025: natts = 3 	-1 -1 -1 
53125: natts = 3 	-1 -1 -1 
53026: natts = 4 	-1 -1 -1 -1 
53126: natts = 4 	-1 -1 -1 -1 
53027: natts = 4 	-1 -1 -1 -1 
53127: natts = 4 	-1 -1 -1 -1 
53028: natts = 5 	-1 -1 -1 -1 -1 
53128: natts = 5 	-1 -1 -1 -1 -1 
53029: natts = 5 	3 -1 -1 -1 -1 
53129: natts = 5 	3 -1 -1 -1 -1 
53030: natts = 5 	-1 -1 -1 -1 -1 
53130: natts = 5 	-1 -1 -1 -1 -1 
53031: natts = 5 	-1 -1 -1 -1 -1 
53032: natts = 1 	-1 
53033: natts = 0 	
53035: natts = 2 	-1 -1 
53036: natts = 2 	-1 -1 
53037: natts = 2 	-1 -1 
53137: natts = 2 	-1 -1 
53038: natts = 5 	-1 -1 -1 -1 -1 
53138: natts = 5 	-1 -1 -1 -1 -1 
53039: natts = 5 	-1 -1 -1 -1 -1 
53139: natts = 5 	-1 -1 -1 -1 -1 
53040: natts = 6 	-1 -1 -1 -1 -1 -1 
53140: natts = 6 	-1 -1 -1 -1 -1 -1 
53041: natts = 6 	-1 -1 -1 -1 -1 -1 
53141: natts = 6 	-1 -1 -1 -1 -1 -1 
53042: natts = 5 	-1 -1 -1 -1 -1 
53043: natts = 5 	-1 -1 -1 -1 -1 
53044: natts = 5 	-1 -1 -1 -1 -1 
53045: natts = 6 	-1 -1 -1 -1 -1 -1 
53145: natts = 6 	-1 -1 -1 -1 -1 -1 
53046: natts = 6 	-1 -1 -1 -1 -1 -1 
53146: natts = 6 	-1 -1 -1 -1 -1 -1 
53047: natts = 3 	-1 -1 -1 
53052: natts = 5 	-1 -1 -1 -1 -1 
53053: natts = 3 	-1 -1 -1 
53153: natts = 3 	-1 -1 -1 
53054: natts = 2 	-1,-1,-1, -1 
53055: natts = 2 	-1,-1,-1, -1 
53056: natts = 2 	{-1} -1 
53057: natts = 2 	{-1},{-1},{-1}, -1 
53058: natts = 2 	{-1,-1},{-1,-1},{-1,-1}, -1 
53059: natts = 2 	-1 -1 
53060: natts = 2 	-1 -1 
53061: natts = 2 	-1 -1 
53062: natts = 2 	-1 -1 
53063: natts = 2 	-1 -1 
53163: natts = 2 	-1 -1 
53064: natts = 2 	-1 -1 
53065: natts = 2 	-1 -1 
53066: natts = 2 	-1 -1 
53067: natts = 5 	-1 -1 -1 -1 -1 
53068: natts = 5 	-1 -1 -1 -1 -1 
53069: natts = 3 	-1 -1 -1 
53070: natts = 5 	-1 -1 -1 -1 -1 
53071: natts = 2 	-1 -1 
53072: natts = 5 	-1 -1 -1 -1 -1 
53073: natts = 5 	-1 -1 -1 -1 -1 
53074: natts = 4 	-1 -1 -1 -1 
53075: natts = 5 	-1 -1 -1 -1 -1 
53076: natts = 6 	-1 -1 -1 -1 -1 -1 
